### PR TITLE
User manager retrieves using username AND email

### LINKED ISSUE #417: Different user retrieved with same username in user management plugin
**Describe the bug / issue**
In user management plugin, a superadmin can search specific user's information based on their username, email or name, the problem happens when there are more than one user with the same username, then if a superadmin try to retrieve the info of one of the users with the same username, it return the first user in the database with the same username and not the the user that the superadmin want to see, even if they search based on name or email.

**To Reproduce**
1. Go to user management as super admin
2. Search a user with the same username repeated in the database
3. Put the email or name in the user management plugin
4. See the error.

**Expected behavior**
We need to filter the search and the retrieve of the user's info in the database, the problem is caused because once a super admin find a user based on any string, the info that is retrieved is based on the username, what is a problem if in the database the username is repeated.

### PR DESCRIPTION
**Pull request by gaperezsa**
# Description
User information retrieval in the User manager page was done using the username of the clicked user. This led to ill behavior when 2 or more users had the same username. Although 2 users with the same usernames is something that shouldn't happen, the recent incident of many empty string users allowed us to notice that the "first" user with an empty string as username would be retrieved no matter which of them was chosen.

Functions involved with the user retrieval were changed so that they also required and used the email along with the username. the respective MongoDB find sentence was changed and the "on-click" behavior was adapted so that the email is also sent when clicked.

Fixes #417 

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
3 instances of a user "juan" were inserted, the first 2 shared username while the first and the third shared email. This were inserted "Manually" into the local database for debugging purposes. Then, through the User Manager, all three instances were chosen to check behavior and the correct instance was retrieved in each case because; while they share email or username with another one, each is unique in their combination of username AND email.

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
image

### CODE REVIEW

File: inginious/frontend/plugins/user_management/pages/api/user_data.py [hunk 1]
```diff
@@ -142,8 +142,9 @@ def API_GET(self):
         """ Get request. Returns data about a user """
         self.check_superadmin_rights()
         username = get_mandatory_parameter(web.input(), "username")
+        email = get_mandatory_parameter(web.input(), "email")
         try:
-            user_data = self.get_user_data(username)
+            user_data = self.get_user_data(username,email)
```
Comment (ferestrepoca) [inline inginious/frontend/plugins/user_management/pages/api/user_data.py]:
```suggestion
            user_data = self.get_user_data(username, email)
```

File: inginious/frontend/plugins/user_management/pages/api/user_data.py [hunk 2]
```diff
@@ -153,14 +154,15 @@ def API_POST(self):
...
+        email = get_mandatory_parameter(user_data, "email")
         collections_manager = CollectionsManagerSingleton.get_instance()
         superadmin_username = self.user_manager.session_username()
 
         if username == superadmin_username:
             return 400, {"error": _("Superadmin can not modify its own data")}
 
         try:
-            user_original_info = self.get_user_data(username)
+            user_original_info = self.get_user_data(username,email)
```
Comment (ferestrepoca) [inline inginious/frontend/plugins/user_management/pages/api/user_data.py]:
```suggestion
            user_original_info = self.get_user_data(username, email)
```

File: inginious/frontend/plugins/user_management/pages/api/user_data.py [hunk 3]
```diff
@@ -171,7 +173,7 @@ def API_POST(self):
         new_username = user_data["new_username"] if username_count > 0 else username
 
         try:
-            user_final_info = self.get_user_data(new_username)
+            user_final_info = self.get_user_data(new_username,email)
```
Comment (ferestrepoca) [inline inginious/frontend/plugins/user_management/pages/api/user_data.py]:
```suggestion
            user_final_info = self.get_user_data(new_username, email)
```

**Review by ferestrepoca — CHANGES_REQUESTED** [review]
LGTM. Just a few code style suggestions and we're done. Thanks!

**Review by VirtualDiegox — APPROVED** [review]
LGTM!

**Review by ferestrepoca — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**